python_utils.py

```diff
@@ -286,17 +290,22 @@ def url_retrieve(source_url, filename=None):
     Returns:
         urlretrieve. The 'urlretrieve' object.
     """
+    import certifi
+    import ssl
```

Comment:
Maybe explain why are the imports here and not at the top of the file.

Reply:
I've moved them to the top of the file